diff --git a/src/llama-kv-cache.cpp b/src/llama-kv-cache.cpp
index d80e8a70b..09f2e05c2 100644
--- a/src/llama-kv-cache.cpp
+++ b/src/llama-kv-cache.cpp
@@ -1954,6 +1954,11 @@ bool llama_kv_cache::state_read_meta(llama_io_read_i & io, uint32_t strm, uint32
             io.read_to(&n_seq_id, sizeof(n_seq_id));
 
             cells.pos_set(i, pos);
+            if (hparams.n_pos_per_embd() > 1) {
+                llama_kv_cell_ext ext;
+                io.read_to(&ext, sizeof(ext));
+                cells.ext_set(i, ext);
+            }
 
             for (uint32_t j = 0; j < n_seq_id; ++j) {
                 llama_seq_id seq_id;
